src/ChatBubble/index.tsx

```diff
@@ -2,41 +2,79 @@ import * as React from "react";
 import styled from "styled-components";
 import Message from "../Message";
 
-const ChatBubbleWrapper = styled.div`
+const BubbleGroupHeader = styled.h5<{ me: boolean }>`
+  margin: 0;
+  margin-bottom: 4px;
+  font-size: 14px;
+  font-weight: 500;
+  color: ${(props) =>
+    props.me
+      ? props.theme.bubbles.mineHeader
+      : props.theme.bubbles.theirHeader};
+  font-family: ${(props) => props.theme.fontFamily};
+`;
+
+const ChatBubbleWrapper = styled.div<{ me: boolean }>`
   overflow: auto;
+  max-width: 420px;
+  margin-left: ${(props) => props.me && "auto"};
 `;
 
-const ChatBubbleUI = styled.div<{ me: boolean }>`
-  background-color: ${props =>
+const ChatBubbleUI = styled.div<{ me: boolean; first?: boolean }>`
+  background-color: ${(props) =>
     props.me
       ? props.theme.bubbles.mineBackground
       : props.theme.bubbles.theirBackground};
-  border-radius: 20px;
+
+  // border-radius: 8px 8px ${(props) => (props.me ? "0 8px" : "8px 0")};
+  border-radius: ${(props) =>
+    props.first ? "8px 8px" : props.me ? "8px 4px" : "4px 8px"}
+    ${(props) => (props.me ? "0 8px" : "8px 0")};
   margin: 1px auto;
-  max-width: 425px;
-  padding: 8px 14px;
-  width: -webkit-fit-content;
-  float: ${props => (props.me ? "right" : "left")};
+  padding: 12px 16px;
+  min-width: fit-content;
+  float: ${(props) => (props.me ? "right" : "left")};
 `;
 
 const MessageUI = styled.p<{ me: boolean }>`
-  color: ${props =>
+  color: ${(props) =>
     props.me ? props.theme.bubbles.mineColor : props.theme.bubbles.theirColor};
-  font-size: ${props => props.theme.bubbles.messageFontSize};
-  font-family: ${props => props.theme.bubbles.messageFontFamily};
+  font-size: ${(props) => props.theme.bubbles.messageFontSize};
+  font-family: ${(props) => props.theme.bubbles.messageFontFamily};
   font-weight: normal;
   margin: 0;
 `;
 
 export interface ChatBubbleProps {
   message: Message;
+  first?: boolean;
+  senderName?: string;
+  showSenderName?: boolean;
 }
 
 export default function ChatBubble(props: ChatBubbleProps) {
+  if (props.first === undefined) {
```

Comment:
I'm not a massive fan of this sort of reassignment. If we can have such dramatic changes based on a single prop being present, should we split into two components?

Reading it a bit more deeply, `props.first` being undefined makes it set to `true` seems a little weird. Can we make it a default value instead?

```
const isFirstMessage = props.first != null ? props.first : true;
```